cdap-api/src/main/java/co/cask/cdap/internal/artifact/Plugin.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.artifact;
+
+import co.cask.cdap.api.artifact.ArtifactVersion;
+import co.cask.cdap.api.templates.plugins.PluginClass;
+import co.cask.cdap.api.templates.plugins.PluginProperties;
+
+import java.net.URI;
+import java.util.Objects;
+
+/**
+ * A container class for holding plugin information.
+ */
+public final class Plugin {
```

Comment:
Is this only needed for serialization/deserialization in mapreduce? If so it shouldn't be in api.